### DISCUSSION TITLE
Built-In Video Playback

### DISCUSSION THREAD

**Discussion by ghost** [musescore/MuseScore#15610] [Ideas]
I'd really love to see a way for video to be played in sync with scores in MuseScore 4. In MS3 I had a setup using JACK Audio along with some other software that with tinkering and some patience usually worked. With the apparent removal of JACK support in MS4, this is --seemingly-- impossible.

I'd love to see a way to play video inside of MS4, or at the very least a way to do it with my old stinky JACK setup. --But frankly I'd rather just have it built in because the workflow introduced 3 more applications open alongside MuseScore and it was frankly just a pain to make happen.--

Thanks,
Sam

**Comment by cfirwin3** [comment]
I will bump this up with a comment.  I think it would be appropriate to merely allow access to the midi timecode output from MuseScore.  I think this would be the most achievable, cross-platform way to resolve sync issues related to score-to-picture AND modularity with other time-aware applications.  Mere access to the timecode allows users to apply applications that can see the timecode and sync them from MuseScore's own playback without having to get caught up in writing a whole Jack interface layer to MuseScore or creating a video play system.  I think if MuseScore development said "We are providing access to the timecode, but nothing else... no slave transport functions, no Jack audio, etc." I honestly don't see that they would need to do anything more on the issue.  The audio output is already accessible as a single stereo channel that can be plugged into session software.  We are just missing the access to the timecode.  It's already there, isn't it?  We just need to be able to see the raw timecode output as a socket that will show up in MTC-aware software.
